### ISSUE TITLE
bug(ui): voting strategies duplicated when saving settings

### ISSUE THREAD

**Issue by bonustrack** [snapshot-labs/sx-monorepo#80]
This happened few time for me but not all the time, when I save settings it sometime duplicate some of the voting strategies, see here for example initially there was only 3 strategies but 2 of them got duplicated:
<img width="908" alt="image" src="https://github.com/snapshot-labs/sx-monorepo/assets/16245250/b4082e72-0d24-4a96-bc6d-c3733493a394">

**Comment by Sekhmet** [comment]
Can you share space address?

**Comment by bonustrack** [comment]
It was on this space: https://snapshotx.xyz/#/sn:0x05702362b68a350c1cae8f2a529d74fdbb502369ddcebfadac7e91da37636947 it doesn't happen all the time

**Comment by Sekhmet** [comment]
I couldn't reproduce this issue. I went through entire history of this space and all seems to make sense. Also it shows correct list of strategies right now on API (and so it did when I run it locally): https://api-1.snapshotx.xyz/?explorerURLState=N4IgJg9gxgrgtgUwHYBcQC4TADpIAR4DOADgIZQIAUAlmOntiAAwAeTArAOxMBMAzADYeAIwEAOUn3ZMoARiikEYgGY9S7HgE4wnACzKww4dP4DtYCsOWkw5Tgk2zbfTgMGbdnRgEo8OfAR4SAgsKAD6hCgATqQoCADmAJ5h1EhgIbiBRNGxCdQIhClp1FD5hJmBkTFx8WUVAL649SD1QA

I also tried to reproduce it on my own space but couldn't after ~5 tries.

Would be good if we can get some space in broken state and do not update settings after that so it's easier to debug.

**Comment by bonustrack** [comment]
Ok weird, I'll let you know if I can reproduce it

**Comment by bonustrack** [comment]
It has happened again, here is the space: https://snapshotx.xyz/#/sn:0x0041ada9121061198b52ae28edeec8ace7c23f2ba8d66938c129af1a2245701c/settings
Here is the tx: 
https://starkscan.co/tx/0x91148b0deac5ae2e6d6e34ddeeaefb505c97a32e483cc2ff8cbc0b7055fc72

**Comment by Sekhmet** [comment]
It looks like pool transactions end up computed twice:
https://logs.betterstack.com/team/60058/tail?q=&s=580084&a=1710352315611022.125000166
https://logs.betterstack.com/team/60058/tail?q=&s=580084&a=1710352340100118.1210000071

Second event seems to be around block [610711](https://starkscan.co/block/610711#events) but this block doesn't have our events - so it seems like RPC node returned pool transactions with delay so we processed it twice (we clear `processedPoolTransactions` after each block so it doesn't prevent it). I don't think it makes sense to keep those indefinitely   (will get too big), but we could keep those for ~100 blocks maybe). We could also have indefinite cache for actually transactions we already processed (meaning those that match our registered events) as second fallback.
